src/components/swap/SwapDetailsDropdown.tsx

```diff
@@ -102,13 +103,18 @@ export default function SwapDetailsDropdown({ trade, syncing, loading, allowedSl
   const theme = useTheme()
   const { chainId } = useWeb3React()
   const [showDetails, setShowDetails] = useState(false)
+  const trace = useTrace()
 
   return (
     <Wrapper>
       <TraceEvent
         events={[BrowserEvent.onClick]}
         name={SwapEventName.SWAP_DETAILS_EXPANDED}
         element={InterfaceElementName.SWAP_DETAILS_DROPDOWN}
+        properties={{
+          ...(trade ? formatEventPropertiesForTrade(trade, allowedSlippage) : {}),
+          ...trace,
+        }}
```

Comment:
this will create a new object and re-render `TraceEvent` repeatedly, right?

Reply:
@JFrankfurt i don't think it was causing much more rendering than the memoized version in my testing, but I memoized just to be safe

Reply:
@JFrankfurt dont think rerenders work that way - creating a new object here will change the prop but rerenders only happen when there are downstream state changes